python/mspasspy/client.py

```diff
@@ -4,10 +4,23 @@
 from mspasspy.db.client import DBClient
 from mspasspy.db.database import Database
 from mspasspy.global_history.manager import GlobalHistoryManager
-from pyspark import SparkConf, SparkContext
-from pyspark.sql import SparkSession
+
+try:
+    from pyspark import SparkConf, SparkContext
+    from pyspark.sql import SparkSession
+
+    hasSpark = True
```

Comment:
Don't use global variables like this. Rename them to something like `__mspasspy_has_pyspark` and `__mspasspy_has_dask_distributed`. Also, for dask, we need to consider the difference between `dask` and `dask.distributed`. Without `dask.distributed`, `dask` could still be used on a single node for parallelism.